pydantic/_internal/_config.py

```diff
@@ -76,7 +77,7 @@ class ConfigWrapper:
     # whether to validate default values during validation, default False
     validate_default: bool
     validate_return: bool
-    protected_namespaces: tuple[str, ...]
+    protected_namespaces: tuple[str | Pattern[str], ...]
```

Comment:
As we use future annotations, we can import `Pattern` from `re` instead

Reply:
Are we already importing `re` elsewhere, and/or is it fast to import? I'm concerned that importing `re` may add to startup time, but 1, we might be using it elsewhere, and 2, even if not I'm not quite sure how the isinstance check against `typing.Pattern` would work if it wasn't somehow importing `re` somewhere

Reply:
Nevermind, we definitely import `re` in lots of internals so that's a non-issue

Reply:
Nice, I didn't realize we had future annotations here. Changed!